pkg/authn/kubernetes/go.mod

```diff
@@ -0,0 +1,47 @@
+module github.com/google/go-containerregistry/pkg/authn/kubernetes
```

Comment:
Can this just live in k8schain rather than having yet another submodule?

Reply:
I suspect (@imjasonh can correct) this was in an effort to deprecate `k8schain`, which I think warrants more discussion.  So if we push out deprecation for another convo, this SGTM.

Reply:
I think it's useful to be able to have a K8s-secret-backed Keychain that doesn't also pull in all of k8schain's dependencies, even though they're much smaller and more well behaved now.